clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5151,13 +5162,17 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // preprocessed inputs and configure concludes that -fPIC is not supported.
   Args.ClaimAllArgs(options::OPT_D);
 
+  bool SkipO =
+      Args.hasArg(options::OPT_fsycl_link_EQ) && ContainsWrapperAction(&JA);
```

Comment:
The fact that early optimizations are disabled for FPGA is the reason why the change isn't going to result in device-side optimization disabling when compiling from source. Does this statement stand?

Reply:
This will not impact device side optimization due to the fact that wrapping never occurs before the device compilation.  The search for wrapper recursively checks inputs and does not check other adjacent toolchains.  This should only impact items after wrapping which in this case is only the host object wrapping for FPGA.